test/gtest/soft_max_new.cpp

```diff
@@ -0,0 +1,408 @@
+#include "test.hpp"
+#include <array>
+#include <cmath>
+#include <iostream>
+#include <iterator>
+#include <limits>
+#include <memory>
+#include <miopen/convolution.hpp>
+#include <miopen/miopen.h>
+#include <miopen/softmax.hpp>
+#include <miopen/tensor.hpp>
+#include <utility>
+#include <algorithm>
+#include <sstream>
+
+#include "driver.hpp"
+#include "get_handle.hpp"
+#include "tensor_holder.hpp"
+#include "verify.hpp"
+#include "random.hpp"
+
+#include <cstdint>
+#include <tuple>
+#include <gtest/gtest.h>
```

Comment:
Probably not all the headers are needed, can you check and remove unused headers?